File: tests/PointTests.cpp
#include <gtest/gtest.h>

#include "clf/Point.hpp"

using namespace clf;

TEST(PointTests, BasicTest) {
  // create a random point of dimension dim
  const std::size_t dim = 5;


  Eigen::VectorXd x = Eigen::VectorXd::Random(dim);
  const Point point0(x);
  EXPECT_NEAR((x-point0.x).norm(), 0.0, 1.0e-14);

  x = Eigen::VectorXd::Random(dim);
  const Point point1(x);
  EXPECT_NEAR((x-point1.x).norm(), 0.0, 1.0e-14);
  EXPECT_NE(point0.id, point1.id);
}

File: tests/PointTests.py
"""! @brief Test the point (clf::Point) python interface"""

import unittest
import random
import numpy as np

import PyCoupledLocalFunctions as clf

class TestPoint(unittest.TestCase):
   """! Test the point (clf::Point) python interface"""
   def test_set_point(self):
      """! Test adding and getting a parameter"""
      dim = 5
      x = np.array([random.uniform(-1.0, 1.0) for i in range(dim)])

      point = clf.Point(x)
      self.assertAlmostEqual(np.linalg.norm(point.x-x), 0.0)
